accepted/2024/improve-support-for-packages-signed-with-trusted-signing.md

```diff
@@ -0,0 +1,155 @@
+be released separately but I can also understand that the NuGet team would want to launch this as a complete feature.
+
+## Motivation
+
+Earlier this year [Trusted Signing](https://learn.microsoft.com/en-us/azure/trusted-signing/) was launched by Microsoft and 
+recently support for this was added to `dotnet sign`. Packages signed with this can be uploaded to the NuGet Gallery. But
+because the certificate is only valid for three days a user will probably always need to update their certificate on their 
+account page. I would like to propose a set of changes to improve the experience for users that sign their NuGet package
+with Trusted Signing.
+
+## Explanation
+
+### Functional explanation
+
+### Step 1: Register EKU of certificate that was added by Trusted Signing
+
+#### How would this be implemented?
+
+The first step would be to store the EKU that is inside the certificate that was used to sign the NuGet package with Trusted
+Signing. The certificate of Trusted Signing is only valid for thee days but it contains an EKU to that can be used to get the
+identity of the owner of the certificate. There are two EKU's in the certificate. One that tells us that it was signed with
+Trusted Signing (`1.3.6.1.4.1.311.97.1.0`) and another one that contains the Public Trust identity (e.g. 
+`1.3.6.1.4.1.311.97.990309390.766961637.194916062.941502583`). More details about this can be found here:
+https://learn.microsoft.com/en-us/azure/trusted-signing/concept-trusted-signing-cert-management. 
+
+I would like to propose to link EKU's to the user in a similar way as that is now done with certificates:
+
+![EKU](images/trusted-signing-eku.png)
+
+But this would require a user to enter that EKU in the interface to link it with their account. We could also start with
+automatically linking the EKU to the user by using the current "certificate upload" interface. The user would need to do the
+following:
+- Extract the `.cer` file of the signing certificate from the NuGet package.
+- Upload the `.cer` file and link it to their account.
+- Upload the NuGet package that was signed with Trusted Signing.
+
+Below is a picture that show how the certificate and EKU would be linked to user before and after the NuGet package is uploaded.
+
+![EKU link](images/trusted-signing-eku-link.png)
+
+This would mean that the certificate is no longer linked to the user and will no longer be shown in the interface because the EKU
+is now linked to the user. That will mean that the user would need to contact the NuGet team to remove the EKU link until updates
+to the interface have been made. But I don't think a lot of people will need this functionality right away.
+
+The EKU will only be linked when the following conditions are true:
+- The package was signed within the last X days where X should be decided by the NuGet team (not sure if this is really necessary).
```

Comment:
I spoke to the team internally. Let's make this value configurable (an appsetting.json value for the related validation job) and set the default value to 30 days. We can change this to a larger number if needed in the future, based on customer feedback.

We selected 30 days based on this data:

99.4% of packages on NuGet.org today have an author sign timestamp within 30 days of the upload time. 
@JonDouglas [ran a poll on Twitter](https://x.com/nuget/status/1836452574452498890) for this question and found this value will work for 93.5% of respondents.

Reply:
I have updated the window to 30 days.